noir-projects/aztec-nr/aztec/src/messages/encryption/aes128.nr

```diff
@@ -85,33 +90,43 @@ fn extract_close_to_uniformly_random_256_bits_from_ecdh_shared_secret_using_pose
      * it's the big end of a 254-bit field element.
      */
 
-    let rand1: Field = poseidon2_hash_with_separator(
-        [shared_secret.x, shared_secret.y],
-        GENERATOR_INDEX__SYMMETRIC_KEY,
-    );
-    let rand2: Field = poseidon2_hash_with_separator(
-        [shared_secret.x, shared_secret.y],
-        GENERATOR_INDEX__SYMMETRIC_KEY_2,
-    );
+    let mut all_bytes: [[u8; 32]; N] = std::mem::zeroed();
+
+    for k in 0..N {
+        std::static_assert(N < 256, "N too large");
```

Comment:
```suggestion
    std::static_assert(N < 256, "N too large");
    for k in 0..N {
```

Reply:
An explanation for 256 might help

Reply:
I can't explain it... I can't think why I put a limit here, so I just got rid of it.